### ISSUE TITLE
[Neovim] Greeted with `Build Error` on compilation after updating to 5.0.0

### ISSUE THREAD

**Issue by Icy-Thought** [latex-lsp/texlab#837]
### Description

After updating texlab + neovim-git ([nightly overlay](https://github.com/nix-community/neovim-nightly-overlay)) to their latest versions, I noticed that texlab had suddenly become unable to compile my `.tex` files to `.pdf` with the help of tectonic and complained about:

```
[ERROR][2023-01-10 16:52:02] .../vim/lsp/rpc.lua:734	"rpc"	"texlab"	"stderr"	"thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: \"SendError(..)\"', /build/texlab-5.0.0-vendor.tar.gz/lsp-server/src/stdio.rs"
[ERROR][2023-01-10 16:52:02] .../vim/lsp/rpc.lua:734	"rpc"	"texlab"	"stderr"	":29:37\nnote: run with `RUST_BACKTRACE=1` environment variable to display a backtrace\n"
[ERROR][2023-01-10 16:52:02] .../vim/lsp/rpc.lua:734	"rpc"	"texlab"	"stderr"	'Error: unexpected message during shutdown: Request(Request { id: RequestId(I32(5)), method: "shutdown", params: Null })\n'
```

### Other relevant info:

**System information:**
1. Distro: NixOS

**Package versions:**
1. Texlab: `5.0.0`
2. Neovim: `v0.9.0-dev-1df2db0`
3. Tectonic: `0.12.0`

`nvim-lspconfig`
```lua
-- ...
texlab = {
    log_level = vim.lsp.protocol.MessageType.Log,
    settings = {
        texlab = {
            auxDirectory = "build",
            bibtexFormatter = "texlab",
            build = {
                executable = "tectonic",
                args = {
                    "%f",
                    "--synctex",
                    "--keep-logs",
                    "--keep-intermediates",
                    "--outdir",
                    "build",
                },
                onSave = true,
            },
            forwardSearch = {
                executable = "zathura",
                args = { "--synctex-forward", "%l:1:%f", "%p" },
            },
            chktex = {
                onEdit = false,
                onOpenAndSave = false,
            },
            diagnosticsDelay = 100,
            formatterLineLength = 120,
            latexFormatter = "latexindent",
            latexindent = { modifyLineBreaks = false },
        },
    },
```

**Comment by f1rstlady** [comment]
Same here on my system after the upgrade to version `5.0`. I get exactly the same error messages.

 1. Texlab `5.0.0`
 2. Neovim `0.8.2`

**Comment by Icy-Thought** [comment]
I do apologize for the late reply, have been quite busy for the past couple of days..

I managed to install texlab from that commit in Nix and applied `RUST_BACKTRACE=1` as you'd specified and stumbled upon a rather weird issue..

`texlab` appears to be struggling with LaTeX documents placed inside git directories and the reason for my suspicion is that it builds the following document alright when placed outside of my git project directory:

```
├──  build
│  ├──  random.aux
│  ├──  random.log
│  ├──  random.out
│  ├──  random.pdf
│  ├──  random.synctex.gz
│  └──  random.toc
└──  random.tex
```

whereas it fails to build the same document when placed inside my git project directory and complains about a missing `build` directory.

Logs:
```
[DEBUG][2023-01-21 13:36:20] .../lua/vim/lsp.lua:1389	"LSP[texlab]"	"client.request"	1	"textDocument/build"	{  textDocument = {    uri = "file:///git-directory/document/random.tex"  }}	<function 1>	1
[DEBUG][2023-01-21 13:36:20] .../vim/lsp/rpc.lua:284	"rpc.send"	{  id = 4,  jsonrpc = "2.0",  method = "textDocument/build",  params = {    textDocument = {      uri = "file:///git-directory/document/random.tex"    }  }}
[DEBUG][2023-01-21 13:36:20] .../vim/lsp/rpc.lua:387	"rpc.receive"	{  id = 5,  jsonrpc = "2.0",  method = "window/workDoneProgress/create",  params = {    token = 2  }}
[DEBUG][2023-01-21 13:36:20] .../vim/lsp/rpc.lua:403	"server_request: callback result"	{  result = vim.NIL,  status = true}
[DEBUG][2023-01-21 13:36:20] .../vim/lsp/rpc.lua:284	"rpc.send"	{  id = 5,  jsonrpc = "2.0",  result = vim.NIL}
[DEBUG][2023-01-21 13:36:20] .../vim/lsp/rpc.lua:387	"rpc.receive"	{  jsonrpc = "2.0",  method = "$/progress",  params = {    token = 2,    value = {      cancellable = false,      kind = "begin",      message = "file:///git-directory/document/random.tex",      title = "Building"    }  }}
[DEBUG][2023-01-21 13:36:20] .../vim/lsp/rpc.lua:387	"rpc.receive"	{  jsonrpc = "2.0",  method = "window/logMessage",  params = {    message = 'error: output directory "build" does not exist',    type = 4  }}
[DEBUG][2023-01-21 13:36:20] .../vim/lsp/rpc.lua:387	"rpc.receive"	{  jsonrpc = "2.0",  method = "$/progress",  params = {    token = 2,    value = {      kind = "end"    }  }}
[DEBUG][2023-01-21 13:36:20] .../vim/lsp/rpc.lua:387	"rpc.receive"	{  id = 4,  jsonrpc = "2.0",  result = {    status = 1  }}
[INFO][2023-01-21 13:36:20] ...lsp/handlers.lua:516	'error: output directory "build" does not exist'
[DEBUG][2023-01-21 13:36:21] .../vim/lsp/rpc.lua:284	"rpc.send"	{  jsonrpc = "2.0",  method = "textDocument/didSave",  params = {    textDocument = {      uri = "file:///git-directory/document/random.tex"    }  }}
[DEBUG][2023-01-21 13:36:21] .../vim/lsp/rpc.lua:387	"rpc.receive"	{  id = 6,  jsonrpc = "2.0",  method = "window/workDoneProgress/create",  params = {    token = 3  }}
[DEBUG][2023-01-21 13:36:21] .../vim/lsp/rpc.lua:403	"server_request: callback result"	{  result = vim.NIL,  status = true}
[DEBUG][2023-01-21 13:36:21] .../vim/lsp/rpc.lua:284	"rpc.send"	{  id = 6,  jsonrpc = "2.0",  result = vim.NIL}
[DEBUG][2023-01-21 13:36:21] .../vim/lsp/rpc.lua:387	"rpc.receive"	{  jsonrpc = "2.0",  method = "$/progress",  params = {    token = 3,    value = {      cancellable = false,      kind = "begin",      message = "file:///git-directory/document/random.tex",      title = "Building"    }  }}
[DEBUG][2023-01-21 13:36:21] .../vim/lsp/rpc.lua:387	"rpc.receive"	{  jsonrpc = "2.0",  method = "window/logMessage",  params = {    message = 'error: output directory "build" does not exist',    type = 4  }}
[INFO][2023-01-21 13:36:21] ...lsp/handlers.lua:516	'error: output directory "build" does not exist'
[DEBUG][2023-01-21 13:36:21] .../vim/lsp/rpc.lua:387	"rpc.receive"	{  jsonrpc = "2.0",  method = "$/progress",  params = {    token = 3,    value = {      kind = "end"    }  }}
[DEBUG][2023-01-21 13:36:21] .../vim/lsp/rpc.lua:387	"rpc.receive"	{  jsonrpc = "2.0",  method = "textDocument/publishDiagnostics",  params = {    diagnostics = {},    uri = "file:///git-directory/document/random.tex"  }}
[INFO][2023-01-21 13:37:23] .../lua/vim/lsp.lua:1861	"exit_handler"	{ {    _on_attach = <function 1>,    attached_buffers = { true },    cancel_request = <function 2>,    commands = {},    config = {      _on_attach = <function 3>,      autostart = true,      capabilities = {        textDocument = {          callHierarchy = {            dynamicRegistration = false          },          codeAction = {            codeActionLiteralSupport = {              codeActionKind = {                valueSet = { "", "quickfix", "refactor", "refactor.extract", "refactor.inline", "refactor.rewrite", "source", "source.organizeImports" }              }            },            dataSupport = true,            dynamicRegistration = false,            isPreferredSupport = true,            resolveSupport = {              properties = { "edit" }            }          },          completion = {            completionItem = {              commitCharactersSupport = true,              deprecatedSupport = true,              documentationFormat = { "markdown", "plaintext" },              insertReplaceSupport = true,              insertTextModeSupport = {                valueSet = { 1, 2 }              },              labelDetailsSupport = true,              preselectSupport = true,              resolveSupport = {                properties = { "documentation", "detail", "additionalTextEdits" }              },              snippetSupport = true,              tagSupport = {                valueSet = { 1 }              }            },            completionItemKind = {              valueSet = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25 }            },            completionList = {              itemDefaults = { "commitCharacters", "editRange", "insertTextFormat", "insertTextMode", "data" }            },            contextSupport = true,            dynamicRegistration = false,            insertTextMode = 1          },          declaration = {            linkSupport = true          },          definition = {            linkSupport = true          },          documentHighlight = {            dynamicRegistration = false          },          documentSymbol = {            dynamicRegistration = false,            hierarchicalDocumentSymbolSupport = true,            symbolKind = {              valueSet = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26 }            }          },          hover = {            contentFormat = { "markdown", "plaintext" },            dynamicRegistration = false          },          implementation = {            linkSupport = true          },          publishDiagnostics = {            relatedInformation = true,            tagSupport = {              valueSet = { 1, 2 }            }          },          references = {            dynamicRegistration = false          },          rename = {            dynamicRegistration = false,            prepareSupport = true          },          semanticTokens = {            augmentsSyntaxTokens = true,            dynamicRegistration = false,            formats = { "relative" },            multilineTokenSupport = false,            overlappingTokenSupport = true,            requests = {              full = {                delta = true              },              range = false            },            serverCancelSupport = false,            tokenModifiers = { "declaration", "definition", "readonly", "static", "deprecated", "abstract", "async", "modification", "documentation", "defaultLibrary" },            tokenTypes = { "namespace", "type", "class", "enum", "interface", "struct", "typeParameter", "parameter", "variable", "property", "enumMember", "event", "function", "method", "macro", "keyword", "modifier", "comment", "string", "number", "regexp", "operator", "decorator" }          },          signatureHelp = {            dynamicRegistration = false,            signatureInformation = {              activeParameterSupport = true,              documentationFormat = { "markdown", "plaintext" },              parameterInformation = {                labelOffsetSupport = true              }            }          },          synchronization = {            didSave = true,            dynamicRegistration = false,            willSave = true,            willSaveWaitUntil = true          },          typeDefinition = {            linkSupport = true          }        },        window = {          showDocument = {            support = true          },          showMessage = {            messageActionItem = {              additionalPropertiesSupport = false            }          },          workDoneProgress = true        },        workspace = {          applyEdit = true,          configuration = true,          semanticTokens = {            refreshSupport = true          },          symbol = {            dynamicRegistration = false,            hierarchicalWorkspaceSymbolSupport = true,            symbolKind = {              valueSet = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26 }            }          },          workspaceEdit = {            resourceOperations = { "rename", "create", "delete" }          },          workspaceFolders = true        }      },      cmd = { "texlab" },      cmd_cwd = "git-directory",      filetypes = { "tex", "plaintex", "bib" },      flags = {        debounce_text_changes = 150      },      get_language_id = <function 4>,      handlers = <1>{},      init_options = vim.empty_dict(),      log_level = 4,      message_level = 2,      name = "texlab",      on_attach = <function 5>,      on_exit = <function 6>,      on_init = <function 7>,      root_dir = "git-directory",      settings = {        texlab = {          auxDirectory = "build",          bibtexFormatter = "texlab",          build = {            args = { "%f", "--synctex", "--keep-logs", "--keep-intermediates", "--outdir", "build" },            executable = "tectonic",            forwardSearchAfter = false,            onSave = true          },          chktex = {            onEdit = false,            onOpenAndSave = false          },          diagnosticsDelay = 100,          formatterLineLength = 120,          forwardSearch = {            args = { "--synctex-forward", "%l:1:%f", "%p" },            executable = "zathura"          },          latexFormatter = "latexindent",          latexindent = {            modifyLineBreaks = false          }        }      },      single_file_support = true,      workspace_folders = <2>{ {          name = "git-directory",          uri = "file:///git-directory"        } },      <metatable> = <3>{        __tostring = <function 8>      }    },    handlers = <table 1>,    id = 1,    initialized = true,    is_stopped = <function 9>,    messages = {      messages = {},      name = "texlab",      progress = { {          cancellable = false,          done = true,          title = "Building"        }, {          cancellable = false,          done = true,          title = "Building"        }, {          cancellable = false,          done = true,          title = "Building"        } },      status = {}    },    name = "texlab",    notify = <function 10>,    offset_encoding = "utf-16",    request = <function 11>,    request_sync = <function 12>,    requests = {},    rpc = {      is_closing = <function 13>,      notify = <function 14>,      request = <function 15>,      terminate = <function 16>    },    server_capabilities = {      completionProvider = {        resolveProvider = true,        triggerCharacters = { "\\", "{", "}", "@", "/", " " }      },      definitionProvider = true,      documentFormattingProvider = true,      documentHighlightProvider = true,      documentLinkProvider = {        resolveProvider = false      },      documentSymbolProvider = true,      executeCommandProvider = {        commands = { "texlab.cleanAuxiliary", "texlab.cleanArtifacts" }      },      foldingRangeProvider = true,      hoverProvider = true,      inlayHintProvider = true,      referencesProvider = true,      renameProvider = {        prepareProvider = true      },      textDocumentSync = {        change = 2,        openClose = true,        save = {          includeText = false        }      },      workspaceSymbolProvider = true    },    stop = <function 17>,    supports_method = <function 18>,    workspace_did_change_configuration = <function 19>,    workspace_folders = <table 2>,    <metatable> = {      __index = <function 20>    }  } }
[DEBUG][2023-01-21 13:37:23] .../vim/lsp/rpc.lua:284	"rpc.send"	{  id = 5,  jsonrpc = "2.0",  method = "shutdown"}
[DEBUG][2023-01-21 13:37:23] .../vim/lsp/rpc.lua:387	"rpc.receive"	{  id = 5,  jsonrpc = "2.0"}
[DEBUG][2023-01-21 13:37:23] .../vim/lsp/rpc.lua:284	"rpc.send"	{  jsonrpc = "2.0",  method = "exit"}
```

**Comment by pfoerster** [comment]
@Icy-Thought

Thanks for the update!

> texlab appears to be struggling with LaTeX documents placed inside git directories and the reason for my suspicion is that it builds the following document alright when placed outside of my git project directory:

This suspicion seems to align with the issue that I fixed in #839. Can you try again with the `v5.1.0` ( which was just released), please? I think this problem should be fixed now.

**Comment by Icy-Thought** [comment]
Just tested and `texlab` now appears to be working as intended! 
Appreciate the fix! :)

> _pushed update to nixpkgs for others to enjoy!_
